### PR TITLE
➖ Drop support for Python 3.6 and 3.7

### PR DESCRIPTION
**Pull request by ricardo-emanuel01**

### CODE REVIEW

**Comment by ricardo-emanuel01** [comment]
There is something else that I need to change to remove support for Python 3.6 and 3.7?

File: pyproject.toml [hunk 1]
```diff
@@ -16,13 +16,14 @@ classifiers = [
...
-    "Programming Language :: Python :: 3.7",
     "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
 ]
 
 [tool.poetry.dependencies]
-python = "^3.6.1"
+python = "^3.8.1"
```
Comment (Kludex) [inline pyproject.toml:26]:
```suggestion
python = ">=3.8.0"
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.8, 3.9, 3.10, 3.11]
```
Comment (Kludex) [inline .github/workflows/test.yml:16]:
```suggestion
        python-version: ["3.8", "3.9", "3.10", "3.11"]
```

**Comment by Kludex** [comment]
Can we add https://github.com/asottile/pyupgrade#as-a-pre-commit-hook ?

**Comment by ricardo-emanuel01** [comment]
I've never used this, please let me know if something needs to be changed

**Comment by Kludex** [comment]
> I've never used this, please let me know if something needs to be changed

I'll set the pre-commit on the pipeline later, but for now, can you do `pip install pre-commit` and then `pre-commit run --all-files` locally, then check the diff, and push?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -18,3 +18,7 @@ repos:
     rev: 20.8b1
     hooks:
       - id: black
+  - repo: https://github.com/asottile/pyupgrade#as-a-pre-commit-hook
+    rev: v3.9.0
+    hooks:
+      - id: pyupgrade
```
Comment (Kludex) [inline .pre-commit-config.yaml:24]:
```suggestion
      - id: pyupgrade
      - args: ["--py38-plus"]
```

**Comment by ricardo-emanuel01** [comment]
Now I'm investigating the tests problem

File: .pre-commit-config.yaml (hunk 3 above)
Comment (ricardo-emanuel01) [reply .pre-commit-config.yaml:24]: With `- args: ["--py38-plus"]` it did not work

**Comment by Kludex** [comment]
You need to add `httpx` on the requirements.

The problem here is that we didn't pin the dependencies. 😅

**Comment by ricardo-emanuel01** [comment]
It didn't work, even after adding httpx. I had to update pytest and pytest-cov, and only after that, I added httpx. Finally, the tests worked.

**Comment by ricardo-emanuel01** [comment]
Now the tests are running and passing

**Comment by Kludex** [comment]
Thanks :)

**Review by Kludex — APPROVED** [review]

**Pull request merged**